config/initializers/devise.rb

```diff
@@ -281,5 +281,5 @@
 
   config.saml_session_index_key = :session_index
 
-  config.allowed_clock_drift_in_seconds = 1.second
+  config.allowed_clock_drift_in_seconds = 5.minutes
```

Comment:
From the incident email thread and slack, can I confirm it resolved our end by this approach? I don't believe this occurred before, so the configuration in the email suggests clock drift was always a risk we have had all this time?
is 5 minutes excessive if had 7 seconds of clock drift?

Reply:
Yes, this resolved the incident.  It is arguably excessive, but given the anecdotal tales of ESXi guest behaviour with no NTP config (out by 30 minutes or more), I've made a judgement call.  My own view is that, if someone is faking or high-jacked SAML responses given the trust relationship, we have bigger problems than extending the window of abuse by 5 mins.

Our own experience contradicts the anecdotal horror stories; we've always been running this `1.second` tolerance and are not aware of any significant outages where the IdP has drifted beyond this, but given we're relying on user-reported issues, rather than granular data from the app itself, I'd be cautious of claiming "we've never seen clock drift before".